### PR TITLE
Added nautobot.extras.forms.NautobotModelForm and nautobot.extras.filters.NautobotFilterSet and refactored throughout codebase

### PR DESCRIPTION
**Pull request by briddo**
### Fixes: #368 partially for now

This adds nautobot.extras.forms. NautobotModelForm and refactors code to to use this class where appropriate. Will implement the filters next.

### CODE REVIEW

**Review by glennmatthews — COMMENTED** [review]
Looks like a nice bit of cleanup! Can you also add some documentation, both in the release-notes (this is a significant enough feature to be called out individually, IMHO) and in the developer docs (not sure exactly where makes the most sense, possibly in best-practices.md or style-guide.md?)

File: nautobot/circuits/forms.py [hunk 1]
```diff
@@ -7,6 +7,7 @@
     CustomFieldFilterForm,
     CustomFieldModelForm,
     CustomFieldModelCSVForm,
+    GenericBaseModelForm,
```
[the remark this answers is not in the document]
Comment (briddo) [reply nautobot/circuits/forms.py:10]: OK yeah will rename throughout

**Review by glennmatthews — APPROVED** [review]
I'm bothered perhaps more than I should be by the fact that the FilterSet doesn't include a corresponding RelationshipFilterSetMixin, but that doesn't actually exist at this point (see #1316). :-)

File: nautobot/circuits/filters.py [hunk 2]
```diff
@@ -102,24 +98,13 @@ def search(self, queryset, name, value):
...
 
 
-class CircuitFilterSet(
-    BaseFilterSet,
-    StatusModelFilterSetMixin,
-    CustomFieldModelFilterSet,
-    TenancyFilterSet,
-    CreatedUpdatedFilterSet,
-):
+class CircuitFilterSet(NautobotFilterSet, NameSlugSearchFilterSet, StatusModelFilterSetMixin, TenancyFilterSet):
```
Comment (glennmatthews) [inline nautobot/circuits/filters.py:107]: Was the addition of `NameSlugSearchFilterSet` here intentional?

File: nautobot/dcim/filters.py [hunk 3]
```diff
@@ -1151,7 +1102,7 @@ def search(self, queryset, name, value):
         return queryset.filter(qs_filter)
 
 
-class CableFilterSet(StatusModelFilterSetMixin, BaseFilterSet, CreatedUpdatedFilterSet, CustomFieldModelFilterSet):
+class CableFilterSet(StatusModelFilterSetMixin, NautobotFilterSet):
```
Comment (glennmatthews) [inline nautobot/dcim/filters.py:1105]: Is the order here significant in some way or can it be changed to `(NautobotFilterSet, StatusModelFilterSetMixin)` to be consistent with the others?

File: nautobot/circuits/filters.py (hunk 2 above)
Comment (briddo) [reply nautobot/circuits/filters.py:107]: Nope! I did this on the train earlier and it was a bit bumpy! Will sort it in the morning!

File: nautobot/dcim/filters.py (hunk 3 above)
Comment (briddo) [reply nautobot/dcim/filters.py:1105]: Good spot

**Pull request merged**